cleanlab/rank.py

```diff
@@ -109,57 +123,335 @@ def get_self_confidence_for_each_label(
+            """
+        )
+
+    # Adjust predicted probabilities
+    if adj_pred_probs:
+        pred_probs = subtract_confident_thresholds(labels, pred_probs)
+
+    # Pass keyword arguments for scoring function
+    input = {"labels": labels, "pred_probs": pred_probs}
+
+    # Calculate scores
+    label_quality_scores = scoring_func(**input)
+
+    return label_quality_scores
+
+
+def score_label_quality_ensemble(
+    labels: np.array,
+    pred_probs_list: List[np.array],
+    method: str = "self_confidence",
+    adj_pred_probs: bool = False,
+    weight_ensemble_members_by: str = "uniform",
+    verbose: int = 0,
+) -> np.array:
+    """Returns label quality scores based on predictions from an ensemble of models.
+
+    This is a function to compute label-quality scores for classification datasets,
+    where lower scores indicate labels less likely to be correct.
+
+    Ensemble scoring requires a list of pred_probs from each model in the ensemble.
+
+    For each pred_probs in list, compute label quality score.
+    Take the average of the scores with the chosen weighting scheme determined by weight_ensemble_members_by.
+    Default weight_ensemble_members_by is "uniform" which computes the simple average of scores.
+
+    Score is between 0 and 1.
+    1 - clean label (given label is likely correct).
+    0 - dirty label (given label is likely incorrect).
+
+    Parameters
+    ----------
+    labels : np.array
+      Labels in the same format expected by the `score_label_quality()` method.
+
+    pred_probs_list : List of np.array (shape (N, K))
+      Each element in this list should be an array of pred_probs in the same format
+      expected by the `score_label_quality()` method
+
+    method : {"self_confidence", "normalized_margin", "confidence_weighted_entropy"}, default="self_confidence"
+      Label quality scoring method. Default is "self_confidence".
+
+      .. seealso::
+        :func:`self_confidence`
+        :func:`normalized_margin`
+        :func:`confidence_weighted_entropy`
+
+    adj_pred_probs : bool, default = True
+      Adj_pred_probs in the same format expected by the `score_label_quality()` method.
+
+    weight_ensemble_members_by : {"uniform", "accuracy"}, default="uniform"
+      Weighting scheme used to aggregate scores from each model:
+        "uniform": Take the simple average of scores
+        "accuracy": Take weighted average of scores, weighted by model accuracy
+
+    verbose : int
+      If 0, no print statements. If 1, print weighting scheme and accuracy of each model.
+
     Returns
     -------
-      self_confidence : np.array (float)
-        Return the holdout probability that each example in pred_probs belongs to its
-        label. Assumes pred_probs is computed holdout/out-of-sample."""
+    label_quality_scores : np.array (float)
+
+    See Also
+    --------
+    self_confidence
+    normalized_margin
+    confidence_weighted_entropy
+    score_label_quality
```

Comment:
I think See Also should only be `score_label_quality`. The other ones are all explained/linked in the `score_label_quality` documentation

Reply:
Agree, removed the others so now it's only `score_label_quality`
3dbbe7054aeae25b8bddd73a53838cce35b06fa9